link/src/main/java/com/stripe/android/link/ui/verification/VerificationDialog.kt

```diff
@@ -15,61 +15,78 @@ import androidx.compose.runtime.setValue
+    NavHost(navController = navController, startDestination = "dialog") {
+        composable("dialog") {
+            var openDialog by remember { mutableStateOf(true) }
 
-    linkAccount.value?.let { account ->
-        if (openDialog) {
-            Dialog(
-                onDismissRequest = {
-                    openDialog = false
-                    onDialogDismissed()
-                }
-            ) {
-                DefaultLinkTheme {
-                    Surface(
-                        modifier = Modifier
-                            .fillMaxWidth()
-                            .border(
-                                width = 1.dp,
-                                color = MaterialTheme.linkColors.disabledText,
-                                shape = MaterialTheme.shapes.medium
-                            ),
-                        shape = MaterialTheme.shapes.medium
+            val injector = requireNotNull(linkLauncher.injector)
+            val linkAccount = linkLauncher.linkAccountManager.linkAccount.collectAsState()
+
+            linkAccount.value?.let { account ->
+                if (openDialog) {
+                    Dialog(
+                        onDismissRequest = {
+                            openDialog = false
+                            verificationCallback(false)
+                        }
                     ) {
-                        Column(Modifier.padding(horizontal = 20.dp)) {
-                            LinkAppBar(
-                                email = account.email,
-                                onCloseButtonClick = {
-                                    openDialog = false
-                                    onDialogDismissed()
-                                }
-                            )
-                            VerificationBody(
-                                headerStringResId = R.string.verification_header_prefilled,
-                                messageStringResId = R.string.verification_message,
-                                showChangeEmailMessage = false,
-                                linkAccount = account,
-                                injector = injector,
-                                onVerificationCompleted = {
-                                    openDialog = false
-                                    onVerificationCompleted()
+                        DefaultLinkTheme {
+                            Surface(
+                                modifier = Modifier
+                                    .fillMaxWidth()
+                                    .border(
+                                        width = 1.dp,
+                                        color = MaterialTheme.linkColors.disabledText,
+                                        shape = MaterialTheme.shapes.medium
+                                    ),
+                                shape = MaterialTheme.shapes.medium
```

Comment:
I'm assuming we're using material theme because we don't want appearance customization to apply here?

Reply:
On Link [I'm extending MaterialTheme](https://github.com/stripe/stripe-android/blob/master/link/src/main/java/com/stripe/android/link/theme/Theme.kt#L36), so the customizations would be applied here

Reply:
As long as it's inside a `DefaultLinkTheme`, which is the case here